inference/models/gaze/gaze.py

```diff
@@ -0,0 +1,334 @@
+        dynamic_axes = {'input': {0: 'batch_size'}, 'output_yaw': {0: 'batch_size'}, 'output_pitch': {0: 'batch_size'}}
+        torch.onnx.export(self, dummy_input, file_path, input_names=["input"],
+                          output_names=['output_yaw', 'output_pitch'],
+                          dynamic_axes=dynamic_axes, verbose=False)
+
+    def forward(self, x):
+        idx_tensor = torch.stack([self._gaze_idx_tensor for i in range(x.shape[0])], dim=0)
+        gaze_pitch, gaze_yaw = super().forward(x)
+
+        pitch_predicted = self._gaze_softmax(gaze_pitch)
+        pitch_predicted = torch.sum(pitch_predicted * idx_tensor, dim=1) * 4 - 180
+        pitch_degree = pitch_predicted * np.pi / 180.0
+
+        yaw_predicted = self._gaze_softmax(gaze_yaw)
+        yaw_predicted = torch.sum(yaw_predicted * idx_tensor, dim=1) * 4 - 180
+        yaw_degree = yaw_predicted * np.pi / 180.0
+
+        return pitch_degree, yaw_degree
+
+
+class GazeOnnxRoboflowCoreModel(OnnxRoboflowCoreModel):
+    """Roboflow ONNX Gaze model.
+
+    This class is responsible for handling the ONNX Gaze model, including
+    loading the model, preprocessing the input, and performing inference.
+
+    Attributes:
+        gaze_onnx_session (onnxruntime.InferenceSession): ONNX Runtime session for gaze detection inference.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initializes the GazeOnnxRoboflowCoreModel with the given arguments and keyword arguments."""
+
+        t1 = perf_counter()
+        super().__init__(*args, **kwargs)
+        # Create an ONNX Runtime Session with a list of execution providers in priority order. ORT attempts to load providers until one is successful. This keeps the code across devices identical.
+        self.log("Creating inference sessions")
+
+        # TODO: convert face detector (TensorflowLite) to ONNX model
+
+        self.gaze_onnx_session = onnxruntime.InferenceSession(
+            self.cache_file("L2CSNet_gaze360_resnet50_90bins.onnx"),
+            providers=[
+                (
+                    "TensorrtExecutionProvider",
+                    {
+                        "trt_engine_cache_enable": True,
+                        "trt_engine_cache_path": TENSORRT_CACHE_PATH,
+                    },
+                ),
+                "CUDAExecutionProvider",
+                "CPUExecutionProvider",
+            ],
+        )
+
+        if REQUIRED_ONNX_PROVIDERS:
+            available_providers = onnxruntime.get_available_providers()
+            for provider in REQUIRED_ONNX_PROVIDERS:
+                if provider not in available_providers:
+                    raise OnnxProviderNotAvailable(
+                        f"Required ONNX Execution Provider {provider} is not availble. Check that you are using the correct docker image on a supported device."
+                    )
+
+        # init face detector
+        self.face_detector = mp.tasks.vision.FaceDetector.create_from_options(
```

Comment:
Is it possible to go ahead and get the FaceDetector into onnx format? If so, let's do that. If not, let's just leave everything as pytorch. One of the main advantage of using onnx is to not require pytorch as a dependancy. So being half in pytorch, half in onnx isn't buying us much.

Reply:
The model itself can be converted to onnx model by using [tf2onnx](https://github.com/onnx/tensorflow-onnx). 
However, in mediapipe, they load the tflite model as an internal object, and hide image preprocessing and postprocessing.
So, even we can convert the model, without knowing how the whole pipeline is working, it's almost impossible to use onnx instead of mediepipe for those models. 
I'd prefer to use mediapipe for now, and let's see whether we can unify everything later.